libs/libc/stdio/lib_getc.c

```diff
@@ -32,3 +32,8 @@ int getc(FAR FILE *stream)
 {
   return fgetc(stream);
 }
+
+int getc_unlocked(FAR FILE *stream)
```

Comment:
Just a note for the future discussion: Since `getc` alywas relies on `fgetc` then maybe we can make it as a defined in stdio.h? Especially POSIX states
```
The getc() function shall be equivalent to fgetc, except that if it is implemented as a macro it may evaluate stream more than once, so the argument should never be an expression with side-effects.
```
The same can be applied for `putc`

Reply:
@pkarashchenko The PR https://github.com/apache/nuttx/pull/1321 move getc/putc these api to c souce, maybe it's because of C++ problem

Reply:
Ok. Strange since spec says explicitly that those can be a macro. I will explore this deeply. Thanks